sdks/go/pkg/beam/registration/registration_test.go

```diff
@@ -155,6 +157,48 @@ func TestRegister_RegistersTypes(t *testing.T) {
 	}
 }
 
+func TestEmitter1(t *testing.T) {
+	Emitter1[int]()
+	if !exec.IsEmitterRegistered(reflect.TypeOf((*func(int))(nil)).Elem()) {
+		t.Fatalf("exec.IsEmitterRegistered(reflect.TypeOf((*func(int))(nil)).Elem()) = false, want true")
+	}
+}
+
+func TestEmitter2(t *testing.T) {
+	Emitter2[int, string]()
+	if !exec.IsEmitterRegistered(reflect.TypeOf((*func(int, string))(nil)).Elem()) {
+		t.Fatalf("exec.IsEmitterRegistered(reflect.TypeOf((*func(int, string))(nil)).Elem()) = false, want true")
+	}
+}
+
+func TestEmitter2_WithTimestamp(t *testing.T) {
+	Emitter2[typex.EventTime, string]()
+	if !exec.IsEmitterRegistered(reflect.TypeOf((*func(typex.EventTime, string))(nil)).Elem()) {
+		t.Fatalf("exec.IsEmitterRegistered(reflect.TypeOf((*func(typex.EventTime, string))(nil)).Elem()) = false, want true")
+	}
+}
+
+func TestEmitter3(t *testing.T) {
+	Emitter3[typex.EventTime, int, string]()
```

Comment:
I just noted coverage is pretty low for the emitters and iterators (12% of the diff), but since the `exec.make*` calls are unexported, that makes it slightly more difficult.

I think even instantiating them manually only loses coverage of the factory closures anyway so we're probably fine if we were to do it that way. This way we'll be testing *everything* that is instantiated by users too.